changelog.d/1465.feature

```diff
@@ -0,0 +1 @@
+Make message edits more compact when possible
```

Comment:
I'd go a bit further here: `Render matrix message edits as diff statements, falling back to asterisk formatted messages` or some such

Reply:
Improved in 8ba51429